discord/client.py

```diff
@@ -573,7 +573,7 @@ def start(self, email, password):
         yield from self.login(email, password)
         yield from self.connect()
 
-    def run(self, email, password):
+    def run(self, email, password, token):
```

Comment:
It seems a bit ugly to have token as an additional arg when it actually should replace the email/password. Either it should probably use a different method, or you should abstract the credentials to a single variable which could either contain an email/password or a token.